File: lc-go/src/listNode.go
package src

type ListNode struct {
	Val  int
	Next *ListNode
}

func NewList(data []int) *ListNode {
	if len(data) == 0 {
		return nil
	}
	var root, tail *ListNode
	for _, k := range data {
		if root == nil {
			root = &ListNode{k, nil}
			tail = root
		} else {
			tail.Next = &ListNode{k, nil}
			tail = tail.Next
		}
	}
	return root
}
func displayList(head *ListNode) {
	for head != nil {
		print(head.Val, "->")
		head = head.Next
	}
	println("nil")
}

File: lc-java/src/main/java/com/wkr/common/ListNode.java
package com.wkr.common;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

/**
 * @Description:
 * @date: 2022/10/9 19:49
 * @author: wangkun
 */
public class ListNode {
    public int val;
    public ListNode next;
    public ListNode(int x) {
        val = x;
    }
    public static ListNode init(IntStream intStream) {
        return init(intStream.boxed().collect(Collectors.toList()));
    }
    public static ListNode init(List<Integer> inList) {
        ListNode head = null, tail = null;
        for (Integer i : inList) {
            ListNode node = new ListNode(i);
            if (head == null) {
                head = node;
            } else {
                tail.next = node;
            }
            tail = node;
        }
        return head;
    }
    public static void show(ListNode head) {
        System.out.println(toListString(head));
    }
    public static String toListString(ListNode head) {
        StringBuilder builder = new StringBuilder();
        while (head != null) {
            builder.append(head.val);
            builder.append("->");
            head = head.next;
        }
        builder.append("null");
        return builder.toString();
    }

    public static void main(String[] args) {
        ListNode.show(ListNode.init(Arrays.asList(1,2,3)));
        ListNode.show(ListNode.init(Arrays.asList()));
    }
}
